src/dos/drive_local.cpp

```diff
@@ -560,6 +566,36 @@ bool localFile::Close() {
 		fhandle = 0;
 		open = false;
 	};
+
+	if (newtime) {
+		// backport from DOS_PackDate() and DOS_PackTime()
+		struct tm tim = { 0 };
+		tim.tm_sec  = (time&0x1f)*2;
+		tim.tm_min  = (time>>5)&0x3f;
+		tim.tm_hour = (time>>11)&0x1f;
+		tim.tm_mday = date&0x1f;
+		tim.tm_mon  = ((date>>5)&0x0f)-1;
+		tim.tm_year = (date>>9)+1980-1900;
+		//  have the C run-time library code compute whether standard time or daylight saving time is in effect.
+		tim.tm_isdst = -1;
+		// serialize time
+		mktime(&tim);
+
+		struct utimbuf ftim;
+		ftim.actime = ftim.modtime = mktime(&tim);
+
+		char fullname[CROSS_LEN];
+		strcpy(fullname, basedir);
+		strcat(fullname, name);
+//		Dos_SpecoalChar(fullname, true);
+		CROSS_FILENAME(fullname);
+		if (utime(fullname, &ftim)) {
+//			extern int errno;
+//			LOG_MSG("Set time failed for %s (%s)", fullname, strerror(errno));
```

Comment:
Comment line 593: either remove or replace with including `<errno.h>`, as is the practice on all modern platforms.
Line 594: I would be ok with uncommenting this line, but if there's some reason to keep it commented (e.g. it spams on output), then place comment as if it was nested inside the block.